src/internal/middleware.js

```diff
@@ -91,5 +93,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     return task
   }
 
+  sagaMiddleware.setContext = (props) => {
```

Comment:
well, one is needed upon creation and latter might be used in a runtime, could be ofc in response for some action but for more complex contexts it would require action to contain ur dependency which could be not serializable fully, which i dont like - if you for example need to pass some event bus with methods on it